src/sietch/distributed_dune.ml

```diff
@@ -155,6 +184,12 @@ let prefetch ({ cache; _ } as t) count i key =
       block_get t key >>= function
       | None -> Async.Deferred.Result.return ()
       | Some (metadata_contents, _) -> (
+        let* metadata_contents =
+          Async.Pipe.fold ~init:""
+            ~f:(fun l r -> Async.return (l ^ r))
```

Comment:
I don't understand this bit. Are we streaming individual metadata files? That doesn't seem necessary since they should be pretty small.

Reply:
I'd rather reuse the same RPC than introduce a new one that specifically does *not* stream files.